dowhy/causal_estimator.py

```diff
@@ -104,57 +86,68 @@ def __init__(
-            else:
-                self._effect_modifier_names = None
-
         # Check if some parameters were set, otherwise set to default values
         self.num_null_simulations = num_null_simulations
         self.num_simulations = num_simulations
         self.sample_size_fraction = sample_size_fraction
         self.confidence_level = confidence_level
         self.num_quantiles_to_discretize_cont_cols = num_quantiles_to_discretize_cont_cols
         # Estimate conditional estimates by default
+        self.need_conditional_estimates = need_conditional_estimates
+
+        self._bootstrap_estimates = None
+        self._bootstrap_null_estimates = None
+
+    def set_data(self, data: pd.DataFrame, treatment_name: List[str], outcome_name: List[str]):
+        """Sets the data for the estimator
+        :param data: data frame containing the data
+        :param treatment: name of the treatment variable
+        :param outcome: name of the outcome variable
+        """
+        # Currently estimation methods only support univariate treatment and outcome
+        self._data = data
+        self._treatment_name = treatment_name
+        self._outcome_name = outcome_name[0]
+        self._treatment = self._data[self._treatment_name]
+        self._outcome = self._data[self._outcome_name]
+
+    def set_effect_modifiers(self, effect_modifier_names: Optional[List[str]] = None):
+        """Sets the effect modifiers for the estimator
+        :param effect_modifiers: Variables on which to compute separate
+            effects, or return a heterogeneous effect function. Not all
+            methods support this currently.
+        """
+        self._effect_modifiers = effect_modifier_names
+        if effect_modifier_names is not None:
+            self._effect_modifier_names = [cname for cname in effect_modifier_names if cname in self._data.columns]
+            if len(self._effect_modifier_names) > 0:
+                self._effect_modifiers = self._data[self._effect_modifier_names]
+                self._effect_modifiers = pd.get_dummies(self._effect_modifiers, drop_first=True)
+                self.logger.debug("Effect modifiers: " + ",".join(self._effect_modifier_names))
+            else:
+                self._effect_modifier_names = []
+        else:
+            self._effect_modifier_names = []
+
         self.need_conditional_estimates = (
-            need_conditional_estimates if need_conditional_estimates != "auto" else bool(self._effect_modifier_names)
+            self.need_conditional_estimates
+            if self.need_conditional_estimates != "auto"
+            else (self._effect_modifier_names and len(self._effect_modifier_names) > 0)
         )
 
-    @staticmethod
-    def get_estimator_object(new_data, identified_estimand, estimate):
+    def get_estimator_object(self, new_data, identified_estimand, estimate, fit_estimator=True):
```

Comment:
fit_estimator is not needed. We can remove the `fit` code from this method and expect refuters to call it explicitly.